backend/src/LibExecution/ProgramTypesToRuntimeTypes.fs

```diff
@@ -171,55 +171,56 @@ module Expr =
       // into fn3 (fn2 (fn1 v a)) b c
       // This conversion should correspond to ast.ml:inject_param_and_execute
       // from the OCaml interpreter
-      let inner = expr1 |> toRT
-      let expr2' =  PT.Pipe.toExpr expr2
-      let rest' = List.map PT.Pipe.toExpr rest
-      List.fold
-        inner
-        (fun prev next ->
-          let rec convert thisExpr =
-            match thisExpr with
-            // TODO: support currying
-
-            | PT.EFnCall (id, fnName, typeArgs, PT.EPipeTarget ptID :: exprs) ->
-              RT.EApply(
-                id,
-                RT.FnName(FQFnName.toRT fnName),
-                List.map DType.toRT typeArgs,
-                prev :: List.map toRT exprs,
-                RT.InPipe pipeID
-              )
-            | PT.EInfix (id, PT.InfixFnCall fnName, PT.EPipeTarget ptID, expr2) ->
-              let (module_, fn, version) = InfixFnName.toFnName fnName
-              let name =
-                PT.FQFnName.Stdlib(
-                  { module_ = module_; function_ = fn; version = version }
-                )
-              let typeArgs = []
-              RT.EApply(
-                id,
-                RT.FnName(FQFnName.toRT name),
-                typeArgs,
-                [ prev; toRT expr2 ],
-                RT.InPipe pipeID
-              )
-            // Binops work pretty naturally here
-            | PT.EInfix (id, PT.BinOp op, PT.EPipeTarget _, expr2) ->
-              match op with
-              | PT.BinOpAnd -> RT.EAnd(id, prev, toRT expr2)
-              | PT.BinOpOr -> RT.EOr(id, prev, toRT expr2)
-            | other ->
-              let typeArgs = [] // TODO: review
-              RT.EApply(
-                pipeID,
-                RT.FnTargetExpr(toRT other),
-                typeArgs,
-                [ prev ],
-                RT.InPipe pipeID
-              )
-          convert next)
-
-        (expr2' :: rest')
+
+      let folder (prev : RT.Expr) (next : PT.Expr) : RT.Expr =
```

Comment:
It seems like this should work on PT.PipeExprs, not PT.Exprs?

Reply:
Yeah, if we decide to process the `PipeExpre` in `eval`, then we definitely should work with `PipeExpr`. I will fix it soon.